contracts/core/LiquidityPool/SportsAMMV2LiquidityPool.sol

```diff
@@ -607,12 +595,8 @@ contract SportsAMMV2LiquidityPool is Initializable, ProxyOwned, PausableUpgradea
         }
     }
 
-    function updateStakingVolume(
-        IStakingThales stakingThales,
-        address _forUser,
-        uint _amount,
-        bool _isDefaultCollateral
-    ) internal {
+    function _updateStakingVolume(address _forUser, uint _amount, bool _isDefaultCollateral) internal {
+        IStakingThales stakingThales = IStakingThales(addressManager.getAddress("StakingThales"));
```

Comment:
The idea was not to have external call each time updateStakingVolume was called.
This will increase the gas cost.

Reply:
ah, because its iterated upon for each user. Yes, you are correct, I think it should be reverted